src/utlis/errorHandler.js

```
@@ -0,0 +1,291 @@
+    message: errorInfo.message,
+    onRetry: errorInfo.canRetry ? retryFn : null,
+  });
+  container.appendChild(errorEl);
+  return errorEl;
+}
+
+/**
+ * Adds validation error styling and message to a specific form field
+ * @param {HTMLInputElement} input - Input element to add error to
+ * @param {string} message - Error message to display
+ */
+function addFieldError(input, message) {
+  const fieldWrapper = findFieldWrapper(input);
+  if (!fieldWrapper) return;
+
+  input.classList.add("field-error");
+  input.setAttribute("aria-invalid", "true");
+
+  const errorEl = renderValidationError(input.name, [
+    { field: input.name, message },
+  ]);
+
+  if (errorEl) {
+    input.setAttribute("aria-describedby", `${input.name}-error`);
+    fieldWrapper.appendChild(errorEl);
+  }
+}
+
+/**
+ * Removes all validation error messages and styling from a form
+ * @param {HTMLFormElement} form - The form element to clear errors from
+ */
+function clearValidationErrors(form) {
+  if (!form) return;
+
+  const errorMessages = form.querySelectorAll(".field-validation-error");
+  errorMessages.forEach((el) => el.remove());
+
+  const errorInputs = form.querySelectorAll(".field-error");
+  errorInputs.forEach((input) => {
+    input.classList.remove("field-error");
+    input.removeAttribute("aria-invalid");
+    input.removeAttribute("aria-describedby");
+  });
+}
+
+/**
+ * Finds the best container element for displaying validation error messages
+ * @param {HTMLInputElement} input - The input element to find a wrapper for
+ * @returns {HTMLElement} The wrapper element or input's parent if no wrapper found
+ */
+function findFieldWrapper(input) {
+  const selectors = [
+    ".form-group",
+    ".field-wrapper",
+    ".input-group",
+    ".form-field",
+    ".input-wrapper",
+  ];
+
+  for (const selector of selectors) {
+    const wrapper = input.closest(selector);
+    if (wrapper) return wrapper;
+  }
+
+  return input.parentElement;
+}
```

This is trying to meet too many selectors and easily become useless with different class names. Ideally we should instead enforce one consistent class in the html and not try to guess every permutation. Right now it can silently failing if structure changes.